tests/yast2_gui/yast2_bootloader2.pm

```diff
@@ -0,0 +1,49 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Package: yast2-bootloader
+# Summary: YaST2 UI test bootloader checks boot code option, kernel parameters,
+#	graphical console, bootloader option, foreign OS and protect boot loader
+#    Make sure those yast2 modules can opened properly. We can add more
+#    feature test against each module later, it is ensure it will not crashed
+#    while launching atm.
+# Maintainer: Zaoliang Luo <zluo@suse.com>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use scheduler 'get_test_suite_data';
+
+sub bootcode_options_tab {
+    my $test_data        = get_test_suite_data();
+    my $bootcode_options = $testapi::distri->get_bootloader_settings();
+    my %ui_defaults      = $bootcode_options->get_ui_defaults();
+    my @errors;
+    foreach my $setting (keys %{$test_data->{bootloader}->{bootcode_options_tab}}) {
+        record_info("$setting", "$setting: $test_data->{bootloader}->{bootcode_options_tab}->{$setting}");
+        my $expected = $test_data->{bootloader}->{bootcode_options_tab}->{$setting};
+        my $actual   = %ui_defaults{$setting};
+        if ($expected ne $actual) {
+            push @errors, "\n   Found unexpected setting for $setting: Expected $expected. got $actual";
+        }
+    }
+    die "@errors" if (@errors);
+}
+
+sub run {
+    my $bootcode_options = $testapi::distri->get_bootloader_settings();
+    select_console 'x11';
+    YaST::Module::open(module => 'bootloader', ui => 'qt');
+    bootcode_options_tab();
```

Comment:
you are missing a verb when you name method, from the test perspective, we just need to call method of the controller which are just business logic, so words like 'tab' or any other thing UI related should not be mentioned.